### PR TITLE
CAMEL-16019: Create another Camel Paho component for MQTT 5 support

### PR DESCRIPTION
**Pull request by tadayosi**
- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md
----

https://issues.apache.org/jira/browse/CAMEL-16019

I chose the new component name as `paho-mqtt5` as it's more explicit and less confusing and otherwise `paho5` seems confusing as it may indicate "Paho v5" instead of "Paho with MQTT v5".

### CODE REVIEW

File: components/camel-paho-mqtt5/pom.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-test-infra-mosquitto</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
```
Comment (davsclaus) [inline components/camel-paho-mqtt5/pom.xml:82]: Do we need this flatten plugin - cant we use global configured instead?

File: components/camel-paho-mqtt5/src/main/java/org/apache/camel/component/paho/mqtt5/PahoComponent.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Endpoint;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.annotations.Component;
+import org.apache.camel.support.DefaultComponent;
+import org.eclipse.paho.mqttv5.client.MqttClient;
+
+/**
+ * Component to integrate with the Eclipse Paho MQTT library.
```
Comment (davsclaus) [inline components/camel-paho-mqtt5/src/main/java/org/apache/camel/component/paho/mqtt5/PahoComponent.java:29]: v5

File: components/camel-paho-mqtt5/pom.xml (hunk 1 above)
Comment (oscerd) [reply components/camel-paho-mqtt5/pom.xml:82]: I guess this is because the original camel-paho has the same plugin in his pom

**Review by oscerd — CHANGES_REQUESTED** [review]
The component seems ok, but we have the same problem we had with rabbitmq/spring-rabbitmq, we need to rename the component class and endpoint class, otherwise the endpoint dsl will have trouble.

**Comment by davsclaus** [comment]
Ah yeah the component name would class, good spotted.

So it can be named something like

PahoMqtt5Component

File: components/camel-paho-mqtt5/pom.xml (hunk 1 above)
Comment (tadayosi) [reply components/camel-paho-mqtt5/pom.xml:82]: Yes, it was leftover from original camel-paho. Will remove. Thanks.

**Comment by tadayosi** [comment]
Thanks guys for your feedback. Second try now.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**